### ISSUE TITLE
NPE when saving terminal

### ISSUE THREAD

**Issue by SquidDev** [SquidDev-CC/CC-Tweaks#110]
```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  TransformerLoader (OpenComputers-MC1.8.9-1.5.22.6.jar)
  SpongeCoremod (spongeforge-1.8.9-1890-4.2.0-BETA-1666.jar)
  TweaksLoadingPlugin (CCTweaks-1.8.9-1.0.1-patched.jar)
Contact their authors BEFORE contacting forge

// This is a token for 1 free hug. Redeem at your nearest Mojangsta: [~~HUG~~]

Time: 8/28/16 9:21 AM
Description: Exception ticking world

java.lang.NullPointerException: Exception ticking world
    at dan200.computercraft.core.terminal.Terminal.writeToNBT(Terminal.java:314)
    at dan200.computercraft.shared.common.ServerTerminal.writeDescription(ServerTerminal.java:96)
    at dan200.computercraft.shared.peripheral.monitor.TileMonitor.writeDescription(TileMonitor.java:217)
    at dan200.computercraft.shared.common.TileGeneric.func_145844_m(TileGeneric.java:192)
    at net.minecraft.server.management.PlayerManager$PlayerInstance.func_151252_a(PlayerManager.java:571)
    at net.minecraft.server.management.PlayerManager$PlayerInstance.func_73254_a(PlayerManager.java:518)
    at net.minecraft.server.management.PlayerManager.func_72693_b(PlayerManager.java:64)
    at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:223)
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:690)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:344)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:605)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:481)
    at java.lang.Thread.run(Thread.java:745)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
    at dan200.computercraft.core.terminal.Terminal.writeToNBT(Terminal.java:314)
    at dan200.computercraft.shared.common.ServerTerminal.writeDescription(ServerTerminal.java:96)
    at dan200.computercraft.shared.peripheral.monitor.TileMonitor.writeDescription(TileMonitor.java:217)
    at dan200.computercraft.shared.common.TileGeneric.func_145844_m(TileGeneric.java:192)
    at net.minecraft.server.management.PlayerManager$PlayerInstance.func_151252_a(PlayerManager.java:571)
    at net.minecraft.server.management.PlayerManager$PlayerInstance.func_73254_a(PlayerManager.java:518)
    at net.minecraft.server.management.PlayerManager.func_72693_b(PlayerManager.java:64)
    at net.minecraft.world.WorldServer.func_72835_b(WorldServer.java:223)

-- Affected level --
Details:
    Level name: world
    All players: 2 total; [EntityPlayerMP['umwn'/329, l='world', x=6192.65, y=62.41, z=3330.71], EntityPlayerMP['Diemetradon'/1132, l='world', x=5781.76, y=65.17, z=2713.58]]
    Chunk stats: ServerChunkCache: 663 Drop: 0
    Level seed: 2020475164307036244
    Level generator: ID 00 - default, ver 1. Features enabled: true
    Level generator options: 
    Level spawn location: 5913.00,75.00,2713.00 - World: (5913,75,2713), Chunk: (at 9,4,9 in 369,169; contains blocks 5904,0,2704 to 5919,255,2719), Region: (11,5; contains chunks 352,160 to 383,191, blocks 5632,0,2560 to 6143,255,3071)
    Level time: 115266388 game time, 101808413 day time
    Level dimension: 0
    Level storage version: 0x04ABD - Anvil
    Level weather: Rain time: 5724 (now: true), thunder time: 5724 (now: true)
    Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: false
Stacktrace:
    at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:690)
    at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:344)
    at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:605)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:481)
    at java.lang.Thread.run(Thread.java:745)

-- System Details --
Details:
    Minecraft Version: 1.8.9
    Operating System: Linux (amd64) version 2.6.32-46-pve
    Java Version: 1.8.0_91, Oracle Corporation
    Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
    Memory: 830615368 bytes (792 MB) / 1721634816 bytes (1641 MB) up to 8563785728 bytes (8167 MB)
    JVM Flags: 4 total; -Xmx8192M -Xincgc -XX:ParallelGCThreads=3 -XX:MaxPermSize=128m
    IntCache: cache: 1, tcache: 1, allocated: 12, tallocated: 94
    FML: MCP 9.19 Powered by Forge 11.15.1.1902 23 mods loaded, 23 mods active
    States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
    UCHIJAAAA   mcp{9.19} [Minecraft Coder Pack] (minecraft.jar) 
    UCHIJAAAA   FML{8.0.99.99} [Forge Mod Loader] (forge-1.8.9-11.15.1.1902-1.8.9-universal.jar) 
    UCHIJAAAA   Forge{11.15.1.1902} [Minecraft Forge] (forge-1.8.9-11.15.1.1902-1.8.9-universal.jar) 
    UCHIJAAAA   OpenComputers|Core{1.5.22.6} [OpenComputers (Core)] (minecraft.jar) 
    UCHIJAAAA   sponge{1.8.9-1890-4.2.0-BETA-1666} [SpongeForge] (spongeforge-1.8.9-1890-4.2.0-BETA-1666.jar) 
    UCHIJAAAA   <CCTweaks ASM>{1.0.1} [<CCTweaks ASM>] (minecraft.jar) 
    UCHIJAAAA   io.github.hsyyid.adminshop{1.8.1} [AdminShop] (AdminShop-1.8.9-1.8.1-universal.jar) 
    UCHIJAAAA   ComputerCraft{1.79} [ComputerCraft] (ComputerCraft1.79.jar) 
    UCHIJAAAA   CCTweaks{1.0.1} [CCTweaks] (CCTweaks-1.8.9-1.0.1-patched.jar) 
    UCHIJAAAA   chunkloaderz{1.0.3} [ChunkLoaderZ] (ChunkLoaderZ-1.8.9-1.0.3.jar) 
    UCHIJAAAA   Dynmap{2.3-15} [Dynmap] (Dynmap-2.3-forge-1.8.9.jar) 
    UCHIJAAAA   io.github.hsyyid.essentialcmds{@project.informativeVersion@} [EssentialCmds] (EssentialCmds-1.8.9-8.1.13-universal.jar) 
    UCHIJAAAA   griefprevention{2.0.0} [GriefPrevention] (griefprevention-1.8.9-2.0.0.66.jar) 
    UCHIJAAAA   IronChest{6.0.125.770} [Iron Chest] (ironchest-1.8.9-6.0.125.770.jar) 
    UCHIJAAAA   JEI{2.28.9.176} [Just Enough Items] (jei_1.8.9-2.28.9.176.jar) 
    UCHIJAAAA   io.github.hsyyid.kits{1.7.4} [Kits] (Kits-1.7.4.jar) 
    UCHIJAAAA   minimalperipherals{0.0} [MinimalPeripherals] (MinimalPeripherals-lite-1.8.9-1.1.jar) 
    UCHIJAAAA   MoreFurnaces{1.5.1} [More Furnaces] (MoreFurnaces-1.5.1-MC1.8.9.jar) 
    UCHIJAAAA   OpenComputers{1.5.22.6} [OpenComputers] (OpenComputers-MC1.8.9-1.5.22.6.jar) 
    UCHIJAAAA   ninja.leaping.permissionsex{2.0-SNAPSHOT} [PermissionsEx] (PermissionsEx-Sponge.jar) 
    UCHIJAAAA   com.nguyenquyhy.spongediscord{1.2.0} [Sponge Discord] (Sponge-Discord-1.2.1-all.jar) 
    UCHIJAAAA   totaleconomy{1.4.1} [Total Economy] (TotalEconomy-v1.4.1.jar) 
    UCHIJAAAA   worldedit{6.1.1} [WorldEdit] (worldedit-forge-mc1.8.9-6.1.1-dist.jar) 
    Loaded coremods (and transformers): 
TransformerLoader (OpenComputers-MC1.8.9-1.5.22.6.jar)
  li.cil.oc.common.asm.ClassTransformer
SpongeCoremod (spongeforge-1.8.9-1890-4.2.0-BETA-1666.jar)
  org.spongepowered.common.launch.transformer.SpongeSuperclassTransformer
TweaksLoadingPlugin (CCTweaks-1.8.9-1.0.1-patched.jar)
  org.squiddev.cctweaks.core.asm.ASMTransformer
    Profiler Position: N/A (disabled)
    Player Count: 2 / 50; [EntityPlayerMP['umwn'/329, l='world', x=6192.65, y=62.41, z=3330.71], EntityPlayerMP['Diemetradon'/1132, l='world', x=5781.76, y=65.17, z=2713.58]]
    Is Modded: Definitely; Server brand changed to 'fml,forge,sponge'
    Type: Dedicated Server (map_server.txt)
```

**Comment by SquidDev** [comment]
Possibly related to [this bug](http://www.computercraft.info/forums2/index.php?/topic/27333-server-crashes-when-trying-to-run-a-program/).

**Comment by SquidDev** [comment]
By the looks of it, this occurs when resizing the terminal whilst writing to NBT. This is a vanilla CC bug in an area that I don't currently touch so I won't fix it.

**Comment by SquidDev** [comment]
According to dan200/ComputerCraft#124 this has been fixed in CC 1.8pr1. See also [this bug](http://www.computercraft.info/forums2/index.php?/topic/27333-server-crashes-when-trying-to-run-a-program/)
